lsp-mode.el

```diff
@@ -4689,15 +4689,21 @@ should return the command to start the LS server."
                 (cons tcp-client-connection cmd-proc)))
    :test? (lambda () (executable-find (cl-first (funcall command-fn 0))))))
 
-(defun lsp-tramp-connection (local-command)
+(defun lsp-tramp-connection (local-command &optional generate-error-file-fn)
   "Create LSP stdio connection named name.
 LOCAL-COMMAND is either list of strings, string or function which
 returns the command to execute."
   (list :connect (lambda (filter sentinel name)
                    (let* ((final-command (lsp-resolve-final-function local-command))
                           ;; wrap with stty to disable converting \r to \n
-                          (wrapped-command (append '("stty" "raw" ";") final-command))
-                          (process-name (generate-new-buffer-name name)))
+                          (process-name (generate-new-buffer-name name))
+                          (wrapped-command (append '("stty" "raw" ";")
+                                                   final-command
+                                                   (list
+                                                    (concat "2>"
+                                                            (or (when generate-error-file-fn
+                                                                  (funcall generate-error-file-fn name))
+                                                                (format "/tmp/%s-stderr" name)))))))
```

Comment:
We should try a little harder to prevent these stderr files from stomping on each other. I suggest something like this instead:

```
(format "/tmp/%s-%04x%04x-stderr" name (random (expt 16 4)) (random (expt 16 4)))
```

It just adds a poor man's uuid to the filename. It's completely portable and works well enough in practice.

Reply:
Thank you, I did it via simple counter because I am not quite sure whether the `%04x` format identifier works on the older emacs versions(we recently had a bug about that).

Reply:
Seems like it should work. 👍